flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/FunctionITCase.java

```diff
@@ -1095,7 +1095,7 @@ public void testTimestampNotNull() {
         TestCollectionTableFactory.initData(sourceData);
 
         tEnv().executeSql(
-                        "CREATE TABLE SourceTable(i INT, ts AS LOCALTIMESTAMP, WATERMARK FOR ts AS ts) "
+                        "CREATE TABLE SourceTable(i INT, ts AS CAST(LOCALTIMESTAMP AS TIMESTAMP(3)), WATERMARK FOR ts AS ts) "
```

Comment:
Is this a drive-by change? Or was this necessary?

Reply:
It was necessary, because the new schema validator is a bit more strict when it comes to precision. Currently, we only support time attributes of precision 3. As documented here: https://ci.apache.org/projects/flink/flink-docs-release-1.12/dev/table/sql/create.html#watermark a couple of locations violated this.